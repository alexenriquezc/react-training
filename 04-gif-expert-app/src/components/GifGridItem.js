import React from 'react'

export const GifGridItem = ({ title, url }) => {
    const rand = Math.random();
    return (
        <div className="card animate__animated animate__fadein" key={rand}>
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
