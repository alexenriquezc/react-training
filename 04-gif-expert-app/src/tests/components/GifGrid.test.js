import { GifGrid } from "../../components/GifGrid"
import { shallow } from 'enzyme';
import React from "react";
import '@testing-library/jest-dom';
import useFetchGifs from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');


describe('Testing on <GifGrid/>', () => {

    const category = 'Samurai X';
    // let wrapper = shallow( <GifGrid category={category}/>);

    test('should shows correctly', () => {
        const wrapper = shallow( <GifGrid category={category}/>);
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        expect(wrapper).toMatchSnapshot();
    });

    test('should shows items when images are loading useFetch', () => {

        const gifs = [{
            id: 'ABC',
            url: 'http://localhost',
            title: 'something'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={category}/>);

    })
    
    
})
