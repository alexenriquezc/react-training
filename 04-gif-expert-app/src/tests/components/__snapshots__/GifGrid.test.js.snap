// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <GifGrid/> should shows correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Samurai X
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
